wormchain/x/wormhole/keeper/msg_server_register_account_as_guardian.go

```diff
@@ -40,18 +44,21 @@ func (k msgServer) RegisterAccountAsGuardian(goCtx context.Context, msg *types.M
 	// we don't allow registration of arbitrary public keys, since that would
 	// enable a DoS vector
 	latestGuardianSetIndex := k.Keeper.GetLatestGuardianSetIndex(ctx)
-	consensusGuardianSetIndex, found := k.GetConsensusGuardianSetIndex(ctx)
-
-	if found && latestGuardianSetIndex == consensusGuardianSetIndex.Index {
-		return nil, types.ErrConsensusSetNotUpdatable
+	consensusGuardianSetIndex, consensusIndexFound := k.GetConsensusGuardianSetIndex(ctx)
+	if !consensusIndexFound {
+		return nil, types.ErrConsensusSetUndefined
 	}
 
-	latestGuardianSet, found := k.Keeper.GetGuardianSet(ctx, latestGuardianSetIndex)
-
-	if !found {
+	latestGuardianSet, guardianSetFound := k.Keeper.GetGuardianSet(ctx, latestGuardianSetIndex)
+	if !guardianSetFound {
 		return nil, types.ErrGuardianSetNotFound
 	}
 
+	// If the size of the guardian set is 1, allow hot-swapping the validator address.
+	if consensusIndexFound && latestGuardianSetIndex == consensusGuardianSetIndex.Index && len(latestGuardianSet.Keys) != 1 {
```

Comment:
At this point `consensusIndexFound` is `true` (checked in line 48) so checking for it is redundant.

Reply:
Agreed that the check is a bit redundant for this if-statement on line 58. That said, I added the check on line 48 for some additional defense in depth to avoid weird edge cases where `consensusGuardianSetIndex` isn't found in the first place (it could fail this if-statement on line 58 but still continue down the logic in the rest of the function).

Curious to hear other thoughts on this.

Reply:
FWIW I have no problems with such redundancy.